libs/store_helper/kernel/ebpf_registry_helper_km.h

```diff
@@ -10,40 +10,31 @@
 #endif
 #define REG_CREATE_FLAGS 0
 
-#ifdef __cplusplus
-extern "C"
-{
-#endif
-
-    NTSTATUS
-    convert_guid_to_string(
-        _In_ const GUID* guid, _Out_writes_all_(string_length) wchar_t* string, size_t string_length);
+typedef _Return_type_success_(NT_SUCCESS(return )) uint32_t ebpf_registry_result_t;
 
-    void
-    close_registry_key(ebpf_registry_key_t key);
+NTSTATUS
+convert_guid_to_string(_In_ const GUID* guid, _Out_writes_all_(string_length) wchar_t* string, size_t string_length);
 
-    _Must_inspect_result_ ebpf_registry_result_t
-    write_registry_value_binary(
-        ebpf_registry_key_t key,
-        _In_z_ const wchar_t* value_name,
-        _In_reads_(value_size) uint8_t* value,
-        size_t value_size);
+void
+close_registry_key(ebpf_registry_key_t key);
 
-    _Must_inspect_result_ ebpf_registry_result_t
-    write_registry_value_ansi_string(
-        ebpf_registry_key_t key, _In_z_ const wchar_t* value_name, _In_z_ const char* value);
+_Must_inspect_result_ ebpf_registry_result_t
+write_registry_value_binary(
+    ebpf_registry_key_t key,
+    _In_z_ const wchar_t* value_name,
+    _In_reads_(value_size) uint8_t* value,
+    size_t value_size);
 
-    _Must_inspect_result_ ebpf_registry_result_t
-    write_registry_value_dword(ebpf_registry_key_t key, _In_z_ const wchar_t* value_name, uint32_t value);
+_Must_inspect_result_ ebpf_registry_result_t
+write_registry_value_ansi_string(ebpf_registry_key_t key, _In_z_ const wchar_t* value_name, _In_z_ const char* value);
 
-    _Must_inspect_result_ ebpf_registry_result_t
-    create_registry_key(
-        ebpf_registry_key_t root_key, _In_z_ const wchar_t* sub_key, uint32_t flags, _Out_ ebpf_registry_key_t* key);
+_Must_inspect_result_ ebpf_registry_result_t
```

Comment:
can we just return ebpf_result_t instead of ebpf_registry_result_t?

Reply:
Converted all low-level registry functions to return ebpf_result_t